packages/mui-material/src/styles/ThemeProviderNoVars.tsx

```diff
@@ -8,11 +8,25 @@ export interface ThemeProviderNoVarsProps<Theme = DefaultTheme> {
   theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
 }
 
+let warnedOnceNoSsr = false;
+
 export default function ThemeProviderNoVars<Theme = DefaultTheme>({
   theme: themeInput,
   ...props
 }: ThemeProviderNoVarsProps<Theme>): React.ReactElement<ThemeProviderNoVarsProps<Theme>> {
   const scopedTheme = THEME_ID in themeInput ? themeInput[THEME_ID] : undefined;
+  if (process.env.NODE_ENV !== 'production') {
+    // @ts-ignore
+    if (process.env.NODE_ENV !== 'test' && props.noSsr && !warnedOnceNoSsr) {
```

Comment:
The test env check seems suspicious. Do we have leaks in our own tests?

```suggestion
    if (props.noSsr && !warnedOnceNoSsr) {
```